interfaces/repo.go

```diff
@@ -80,6 +80,112 @@ func (r *Repository) AddInterface(i Interface) error {
 	return nil
 }
 
+// QueryOptions describes options for QueryInterfaces.
```

Comment:
The terminology in the repository is getting a bit wild.. a few hints:

- We have tons of "queries" in the repository already. Why is that different?
- What is the difference between Info and MetaData? Note we just fixed that in the client.

I think we should change some of the existing API to also talk about "connections", but we don't have to do that now so we can both focus on more interesting problems.  That said, I'll try to provide some ideas below for how to not make the API too wild for now either.